src/plugins/issue-pr-link/index.js

```diff
@@ -0,0 +1,168 @@
+/**
+ * @fileoverview Comment on issues when PRs are created/edited to fix them
+ * @author ESLint GitHub Bot Contributors
+ */
+
+"use strict";
+
+//-----------------------------------------------------------------------------
+// Type Definitions
+//-----------------------------------------------------------------------------
+
+/** @typedef {import("probot").Context} ProbotContext */
+
+//-----------------------------------------------------------------------------
+// Helpers
+//-----------------------------------------------------------------------------
+
+// Regex to find issue references in PR titles
+// Matches patterns like: "Fix #123", "Fixes #123", "Closes #123", "Resolves #123"
+const ISSUE_REFERENCE_REGEX = /(?:fix|fixes|close|closes|resolve|resolves)\s+#(\d+)/giu;
+
+// Maximum number of issues to comment on per PR to prevent abuse
+const MAX_ISSUES_PER_PR = 3;
+
+/**
+ * Extract issue numbers from PR title
+ * @param {string} title PR title
+ * @returns {number[]} Array of issue numbers
+ * @private
+ */
+function extractIssueNumbers(title) {
+    const matches = [];
+    let match;
+
+    // Reset regex lastIndex to ensure we start from the beginning
+    ISSUE_REFERENCE_REGEX.lastIndex = 0;
+
+    while ((match = ISSUE_REFERENCE_REGEX.exec(title)) !== null && matches.length < MAX_ISSUES_PER_PR) {
+        const issueNumber = parseInt(match[1], 10);
+        if (!matches.includes(issueNumber)) {
+            matches.push(issueNumber);
+        }
+    }
+
+    return matches;
+}
+
+/**
+ * Create the comment message for the issue
+ * @param {string} prUrl URL of the pull request
+ * @param {string} prTitle Title of the pull request
+ * @param {string} prAuthor Author of the pull request
+ * @returns {string} comment message
+ * @private
+ */
+function createCommentMessage(prUrl, prTitle, prAuthor) {
```

Comment:
```suggestion
function createCommentMessage(prUrl, prAuthor) {
```

I think that `prTitle` is unused here.